### DISCUSSION TITLE
How to use OXC in VS Code with nested config files?

### DISCUSSION THREAD

**Discussion by daronspence** [oxc-project/oxc#6333] [Q&A] [answered]
Hello!

New to a lot of JS toolchain stuff so apologies if this is a dumb question. 

I have a Laravel monorepo with 8 different apps. On our customer facing application, I want to enable OXC for linting of JS/Vue files. The documentation describes OXC as "no config" but I'm having issues making any of it work at all.

I installed the extension into VS Code. I do _NOT_ have the Eslint extension enabled. Is that also required? Do I need an extra extension to get the squigglies to appear?

If I run `npx oxlint@latest` from the root of my customer applicatin, it does work as expected, but I get zero feedback `onType` or `onSave` when modifying `.vue` or `.js` files.

The `oxc_language_server` output has no errors and even after enabling tracing in the extension settings and restarting, I see nothing in the trace output either. My config file is located at `/Users/daron/Work/project/Client-Application/.oxlintrc.json` but I have the `project` folder open at the root of my workspace. If I open just the `Client-Application` as a standalone project, it still doesn't work. 

Here is my `oxlintrc.json` config:

```json
{
    "env": {
        "browser": true
    },
    "globals": {
        "AppData": "readonly"
    },
    "settings": {},
    "rules": {
        "eqeqeq": "warn",
        "import/no-cycle": "error"
    }
}
```

I tried writing the following code in a `.vue` component and expected to at least see some errors re: the `==` usage. Unfortuantely, I see no errors. If I run the CLI command, I do see those lines in the output.

```js
if (1 == 2) {
    console.log('test');
}
const thing = (testing, unused) => {
    if (testing == 1) {
        console.log('hi me');
        unused--;
        debugger;
    }
}
```

Happy to provide more details. I saw the presentation at Vite Conf and wanted to give it a shot!

**Comment by markEHVN** [comment]
I also installed oxc extension but I cannot use it. 😭 I created a question [at here](https://github.com/oxc-project/oxc/discussions/7258).

**Comment by nrayburn-tech** [comment]
Nested configuration isn’t supported yet. It is listed as a feature expected to be available before the Oxlint beta https://github.com/oxc-project/oxc/issues/5653.

**Comment by Sysix** [comment, marked as answer]
I think you have a problem with your VSCode Setup.

As a default the oxc plugin looks at a file named `.eslintrc`. You need to override it in your `settings.json`:

`{
"oxc.enable": true,
"oxc.configPath": "oxlintrc.json"
}`

⚠️  You need to restart VSCode to apply the changes

**Reply by markEHVN** [reply]
yep, I try restart vscode but after I restarted, some oxc settings have been apply but oxc still not working 😢

**Comment by daronspence** [comment]
I went ahead and marked @Sysix as the answer. Idk if there were updates to the plugin since I posted this over a month ago (likely) but coming back to it, things seem to work more "out of the box" (without specifying any config, actually). I was able to specify a custom oxc config file and after a restart of VS Code, the changes for the linter were displayed directly in the editor.

Looking forward to the continued development of this project and everything else from `void(0)` !

**Reply by nrayburn-tech** [reply]
There have been some updates and there’s definitely more to come. Oxlint isn’t in beta yet, so there’s definitely plenty of things to keep working through for an improved experience. Both with the CLI directly and the editor integrations.
